### DISCUSSION TITLE
Popups are nice, add an option to have Rounded Corners as that's more appealing and provides a distinct look

### DISCUSSION THREAD

**Discussion by egvijayanand** [CommunityToolkit/Maui#352] [New Feature Discussions]
Popups are nice, add an option to have Rounded Corners as that's more appealing and provides a distinct look and feel.

 And now almost every platform has implemented rounded corners in their UI shell including Windows.

**Comment by VladislavAntonyuk** [comment]
we need to check if the platform API supports rounded corners.
By the way, you can add it manually:
1. Set the Transparent background color of the Popup
2. Add rounded corners to your view

**Reply by TheCodeTraveler** [reply]
If we are able to do it, we should ensure `IPopup` inherits from [`ICornerElement`](https://github.com/dotnet/maui/blob/62580046e3d99ec1ed5aa0bfab962dc5cafc70c1/src/Controls/src/Core/ICornerElement.cs#L3)

**Reply by pictos** [reply]
And how about using the Border control? Not sure if that's possible TBH

**Reply by GeorgeLeithead** [reply]
In the proposed `AvatarView` that I'm contributing to, the parent control is `Border` and we are also inheriting from `ICornerElement`. It all appears to work well, when setting the `StrokeShape` (of the `Border`) to a `RoundRectangle` and then defining the `CornerRadius`.  Only real issue I have found so far, is that clipping images on Windows has an issue [Issue in clipping images in Border](https://github.com/dotnet/maui/issues/6986).

**Comment by egvijayanand** [comment]
Have another question, have used the single instance of XCT Popup that does not return any value, like a busy indicator, and the `Dismiss()` method would hide the popup from View.

Whereas when the same scenario is attempted with MAUI Popup, after invoking the near equivalent `Close()` method the instance gets disposed of.

Is this intentionally modified behavior or would be addressed in the later previews/GA version?

**Comment by egvijayanand** [comment]
Surrounded my Popup content with a Frame with some CornerRadius and Shadow property set to true.

But since Popup inherits from Element and not VisualElement, hence no BackgroundColor property available to modify it.

![Screenshot_20220408-000242.png](https://user-images.githubusercontent.com/81947404/162273262-d2b76e33-9ac4-48eb-910e-9a8043eba536.png)

Kindly look into it.

**Reply by VladislavAntonyuk** [reply]
Use Color instead of BackgroundColor: https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/Views/Popups/TransparentPopup.xaml

**Reply by egvijayanand** [reply]
It works. The background is transparent now.

Can this be made as a built-in feature supported by inheriting Popup from ICornerElement also so that adding an additional view into the Visual Tree can be avoided, Frame in this case?

**Reply by pictos** [reply]
@egvijayanand did you try using the `Border` control?

**Reply by egvijayanand** [reply]
Using `Border` control pushes the Content with CornerRadius into the background and is not appealing as using Frame.

**Comment by egvijayanand** [comment]
Rounded corners not working with the .NET MAUI GA release and its compatible v1.0.0 release of CommunityToolkit.Maui package.

The same code worked fine in the earlier version providing a rounded corner effect for the Popup.

@jfversluis @pictos @brminnick

**Reply by bijington** [reply]
@egvijayanand if they are not working can you please open an issue from this discussion?

**Reply by egvijayanand** [reply]
Sure.

**Reply by bijington** [reply]
Great thanks

**Comment by Orgbrat** [comment]
Have we determined how to properly get rounded corners with MCT popups?

**Reply by egvijayanand** [reply]
* Set the Popup **Color** (=BackgroundColor property) to Transparent.
* Then use either **Frame** / **Border** to enclose the UI content with **Rounded Corners**

**Comment by CliffAgius** [comment]
Closed as answered
